### ISSUE TITLE
File ACLs get lost when pushing to instance

### ISSUE THREAD

**Issue by roosterfish** [canonical/lxd#12032]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 22
 * The output of "lxc info" or if that fails:
   * LXD version: snap latest/edge

# Issue description

When trying to push ACL applied files to instances (both containers and VMs) the headers get lost. Within the instance you cannot see the ACL anymore.

This issue was discovered during work on https://github.com/canonical/lxd/issues/11901.

# Steps to reproduce

 1. `touch foo`
 2. `setfacl -m u:1234:rwx foo`
 3. `lxc launch ubuntu:jammy c1`
 4. `lxc file push foo c1/root/foo`
 5. `lxc exec c1 apt install acl`
 6. `lxc exec c1 getfacl foo`

**Comment by tomponline** [comment]
Hrm, yeah, I'm not sure if ACLs from the host should be stripped as its pushed into the instance.

What happens if you do `cp` on a file normally?

**Comment by roosterfish** [comment]
It feels like all of them have a flag to explicitly preserve those:

* `cp -p`
* `scp -p`
* `rsync --acls`

Maybe it's more a feature request to have something like `lxc file push --acls` to preserve those.

**Comment by tomponline** [comment]
Yes I think so, I'll reclassify this as a feature and we can leave this for now.
